cli/unregister/unregister.go

```diff
@@ -151,28 +154,51 @@ func DeleteHorizonNode(removeNodeUnregister bool, deepClean bool, timeout int) e
 
 // remove local db, policy files and all the service containers
 func DeepClean() error {
+
+	// detect the node type
+	nodeType := persistence.DEVICE_TYPE_DEVICE
+	if _, err := rest.InClusterConfig(); err == nil {
+		nodeType = persistence.DEVICE_TYPE_CLUSTER
+	}
+
 	// get message printer
 	msgPrinter := i18n.GetMessagePrinter()
 
 	msgPrinter.Printf("Starting external deep clean ...")
 	msgPrinter.Println()
-	cliutils.Verbose(msgPrinter.Sprintf("Stopping horizon..."))
-	cliutils.RunCmd(nil, "systemctl", "stop", "horizon.service")
 
-	msgPrinter.Printf("Deleting local horizon DB...")
-	msgPrinter.Println()
-	cliutils.RunCmd(nil, "bash", "-c", "rm -f /var/horizon/*.db")
-	cliutils.RunCmd(nil, "bash", "-c", "rm -Rf /etc/horizon/policy.d/*")
+	if nodeType == persistence.DEVICE_TYPE_CLUSTER {
+		msgPrinter.Printf("Deleting local horizon DB...")
+		msgPrinter.Println()
+		cliutils.RunCmd(nil, "bash", "-c", "rm -f /var/horizon/*.db")
+		cliutils.RunCmd(nil, "bash", "-c", "rm -Rf /etc/horizon/policy.d/*")
+
+		// kill anax inside the agent container, it will get restarted by the nax.service script
```

Comment:
Does anax restart even after Lily's change to the anax.service? I dont think it will.

Reply:
Yes, it still brings up anax when it is killed by kill command. Lily's fix was for pod shutdown case only.